fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -6063,4 +6068,92 @@ private void assertIndexEntryPrimaryKeyTuples(Set<Tuple> primaryKeys, RecordCurs
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            context.commit();
+        }
+        Tuple conflictTuple = Tuple.from(100, 100, 100);
+        try (final FDBRecordContext context = fdb.openContext()) {
+            FDBRecordStore.Builder builder = recordStore.asBuilder()
+                    .setContext(context)
+                    .setMetaDataProvider(recordStore.getMetaDataProvider());
+
+            final FDBRecordStore store = builder.createOrOpen(FDBRecordStoreBase.StoreExistenceCheck.NONE);
+
+            // Obtain lock
+            final AgilityContext agile = AgilityContext.agile(context, TimeUnit.SECONDS.toMillis(4), 100_000);
+            final FDBDirectory directory = new FDBDirectory(store.indexSubspace(index), index.getOptions(),
+                    null, null, true, agile);
+
+            final Lock lock = directory.obtainLock(IndexWriter.WRITE_LOCK_NAME); // will also flush lock
+
+            // Read/write the conflicting value (without committing, yet)
+            testConflictWithLockAgileSet(agile, conflictTuple);
+
+            // Cause the conflict
+            testConflictWithLockCauseConflict(conflictTuple);
+
+            // finalize - imitating the Lucene flow (close lock, close directory, flush agility-context, then close user-context)
+            boolean gotException = false;
+            try {
+                try {
+                    lock.close();
+                } catch (RuntimeException ex) {
+                    gotException = true;
+                }
+                try {
+                    directory.close();
+                } catch (RuntimeException ex) {
+                    gotException = true;
+                }
+                agile.flushAndClose();
+                context.commit();
+            } catch (RuntimeException ex) {
+                gotException = true;
+            } finally {
+                assertTrue(gotException);
+            }
+        }
+
+        // ensure that a new lock is possible
+        TestRecordsTextProto.SimpleDocument doc3 = createSimpleDocument(1623L, "Salesmen jokes are funnier", 2);
+        try (final FDBRecordContext context = fdb.openContext()) {
+            FDBRecordStore.Builder builder = recordStore.asBuilder()
+                    .setContext(context)
+                    .setMetaDataProvider(recordStore.getMetaDataProvider());
+
+            final FDBRecordStore store = builder.createOrOpen(FDBRecordStoreBase.StoreExistenceCheck.NONE);
+            store.saveRecord(doc3);
+            context.commit();
+        }
+    }
+
+    private void testConflictWithLockAgileSet(AgilityContext agile1, Tuple conflictTuple) {
+        agile1.accept(aContext -> {
+            byte [] conflictKey = path.toSubspace(aContext).pack(conflictTuple);
+            final Transaction tr = aContext.ensureActive();
+            tr.get(conflictKey).thenApply(val -> null).join();
```

Comment:
Why have `.thenApply(val -> null)` here and below?

Reply:
Probably a leftover. Removed. 
✅